SCons/Tool/compilation_db.xml

```diff
@@ -0,0 +1,130 @@
+        ]>
+
+<sconsdoc xmlns="http://www.scons.org/dbxsd/v1.0"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="http://www.scons.org/dbxsd/v1.0 http://www.scons.org/dbxsd/v1.0/scons.xsd">
+
+    <tool name="compilation_db">
+        <summary>
+            <para>
+                Sets up &b-CompilationDatabase; builder which generates a clang tooling compatible compilation database.
+            </para>
+        </summary>
+        <sets>
+            <item>COMPILATIONDB_COMSTR</item>
+            <item>__COMPILATIONDB_UACTION</item>
+            <item>__COMPILATIONDB_UTARGET</item>
+            <item>__COMPILATIONDB_USOURCE</item>
+            <item>__COMPILATIONDB_ENV</item>
+            <item>COMPILATIONDB_USE_ABSPATH</item>
+        </sets>
+    </tool>
+
+    <builder name="CompilationDatabase">
+        <summary>
+            <para>
+                The &b-CompilationDatabase; builder writes a
+                <ulink url="https://clang.llvm.org/docs/JSONCompilationDatabase.html">clang formatted compilation
+                    database
+                </ulink> which is consumed by a number of clang tools, editors, and other tools.
+            </para>
+            <para>
+                If you don't specify any files, the builder will default to <filename>compile_commands.json</filename>.
+            </para>
+            <para>
+                If you specify a single file as below
+                <programlisting language="python">
+env.CompilationDatabase('my_output.json')
+                </programlisting>
+                SCons will automatically use that as the target file.
+                If you specify more than one source, the source list will be ignored.
+            </para>
+            <para>
+                You should not specify source files. The &b-CompilationDatabase; builder instruments SCons to collect them from all
+                the C, C++, assembly source/target pairs.
+            </para>
+            <para>
+                NOTE: You must load the compilation_db tool prior to specifying any part of your build or some source/target
+                files will not show up in your output file.
+            </para>
+            <para>
+                New in &SCons; 4.0.0
+            </para>
+        </summary>
+    </builder>
+
+    <cvar name="COMPILATIONDB_COMSTR">
+        <summary>
+            <para>
+                The string displayed when CompilationDatabase builder's action is run.
+            </para>
+        </summary>
+    </cvar>
+
+    <cvar name="COMPILATIONDB_USE_ABSPATH">
+        <summary>
+            <para>
+                This is a boolean flag to instruct &b-link-CompilationDatabase; to
+                write the <literal>file</literal> and <literal>target</literal> members
+                in the target file with absolute or relative path.
+            </para>
+            <para>
+                The default value is False.
```

Comment:
Reword?   _boolean... with absolute or relative path. ... The default value is False._   Text should indicate what False means (yes, you can deduce from name of var, but...).  I'd propose (considering the previous comment also): "This is a boolean flag to instruct the &b-link-CompilationDatabase; builder to record the <literal>file</literal> and <literal>target</literal> members in the compilation database using absolute paths. The default value is <constant>False</constant>, indicating relative paths will be used.